server.js

```diff
@@ -34,29 +36,41 @@ const userMessageTimes = {};
 
 /* Handle incoming messages */
 bot.on("message", msg => {
-    const query = msg.content.substr(commandChar.length).split(" ");
-    const command = query[0].toLowerCase();
-    const parameter = query.length > 1 ? query.slice(1).join(" ") : "";
+    // generate RegExp pattern for message parsing
+    // Example: ^!(card|price) ?(.*)$|!(card|price) ?([^!]*)(!|$)
+    const charPattern = _.escapeRegExp(commandChar);
+    const commandPattern = charPattern+'('+Object.keys(handlers).map(_.escapeRegExp).join('|')+')';
+    const regExpPattern = `^${commandPattern} ?(.*)$|${commandPattern} ?([^${charPattern}]*)(${charPattern}|$)`;
+
+    const queries = msg.content.match(new RegExp(regExpPattern, 'ig')) || [];
```

Comment:
Generating the pattern could be moved out of the message handler.

We could use the capture groups to extract command and parameter here, then we don't have to do further string processing down below.